soc/xtensa/intel_adsp/common/include/intel_adsp_hda.h

```diff
@@ -207,9 +207,14 @@ static inline uint32_t intel_adsp_hda_get_buffer_size(uint32_t base,
  * @param regblock_size Register block size
  * @param sid Stream ID
  */
-static inline void intel_adsp_hda_enable(uint32_t base, uint32_t regblock_size, uint32_t sid)
+static inline void intel_adsp_hda_enable(uint32_t base, uint32_t regblock_size,
+					 uint32_t sid, bool set_fifordy)
 {
-	*DGCS(base, regblock_size, sid) |= DGCS_GEN | DGCS_FIFORDY;
+	*DGCS(base, regblock_size, sid) |= DGCS_GEN;
+
+	if (set_fifordy) {
+		*DGCS(base, regblock_size, sid) |= DGCS_FIFORDY;
+	}
```

Comment:
Drive-by style nitpick.  This whole function exists to abstract out one register write.  It was arguably needless to begin with, but now it sort of looks silly.  Maybe just add something like this in the outer function:

```c
#define IS_HOST_XFER(dir) ((dir) == HOST_TO_MEMORY || (dir) == MEMORY_TO_HOST)


void *dgcs_reg = DGCS(cfg->base, cfg->regblock_size, channel);
uint32_t val = DGCS_GEN | (IS_HOST_XFER(cfg->direction) ? DGCS_FIFORDY : 0)), reg)

sys_write32(dgcs_val, dgcs_reg);
```